src/helpers/factory.ts

```diff
@@ -1,29 +1,31 @@
-import { NextRequest, NextResponse } from "next/server";
-import { withQuery } from "ufo";
+import { NextRequest, NextResponse } from 'next/server'
+import { withQuery } from 'ufo'
 
 export const createHandler = (region: string) => {
   return async (req: NextRequest) => {
-    const body = await req.json();
-    const { url } = body;
-    if (!url) return;
+    const body = await req.json()
+    const { url } = body as { url: string }
+    if (!url) return
 
     try {
-      const startAt = Date.now();
+      const startAt = Date.now()
 
-      await fetch(withQuery(url, { __pong: startAt }));
+      const urlWithProtocol = url.startsWith('http') ? url : `https://${url}`
+
+      await fetch(withQuery(urlWithProtocol, { __pong: startAt }))
```

Comment:
Maybe we can just use `withHttps` from https://github.com/unjs/ufo#withhttps. It's a very handy package for handle URL related stuff btw.